types/history/index.d.ts

```diff
@@ -48,6 +48,7 @@ export namespace History {
     export type Pathname = string;
     export type Search = string;
     export type TransitionHook = (location: Location, callback: (result: any) => void) => any;
+    export type TransitionPromptHook = (location: Location, action: Action) => string | boolean;
```

Comment:
To be more precise, returning `string | false | void` will be better since **TransitionManager** only check returned value is `string` or `false`.

(You don't need to consider the existing `TransitionHook` type because it doesn't used in `history@4` and exists to support back-compat of outdated type definitions in the DT which are not updated to the `history@4`)

Reply:
It may be more precise, but probably only due to an implementation detail of that function (checking directly against `false`).

I think in practice you're much more likely to have a function that just returns a boolean.


```ts
function isFormDirty(): boolean {
     return isEqual(originalModel, newModel);
}

history.block(() => isFormDirty()); // in real implementation obviously you'd do something else with this
```

```ts
history.block(() => { 
   if(isFormDirty()) {
     return false;
  }
});
```

I'll definitely add the "void", and I'll also change to `false` if  you feel strongly about it.

Reply:
On second thought -- maybe forcing it to return false (and making the code be more explicit) is a good thing -- otherwise you might assume that false is the same as void.

Reply:
Updated